src/xo-app/vm/tab-advanced.js

```diff
@@ -113,6 +116,51 @@ class AffinityHost extends Component {
   }
 }
 
+@connectStore(() => ({
+  container: createGetObject((_, { vm }) => vm.$container)
+}))
+class CoresPerSocket extends Component {
+  _updateXenCoresPerSocketValue = createSelector(
+    (maxCoresPerSocket, vCPUs) => getCoresPerSocketPossibilities(maxCoresPerSocket, vCPUs),
+    options => {
+      const coresPerSocket = this.props.vm.coresPerSocket
+
+      if (coresPerSocket !== undefined && !includes(options, +coresPerSocket)) {
+        editVm(this.props.vm, { coresPerSocket: null })
+      }
+    }
+  )
+
+  render () {
+    const {
+      container,
+      vm
+    } = this.props
+    const maxCoresPerSocket = container.cpus.cores
```

Comment:
Both have cpus, if maxCoresPerSocket is undefined it handled in the getCoresPerSockets.